src/Data/Text.hs

```diff
@@ -535,9 +541,15 @@ null (Text _arr _off len) =
 #if defined(ASSERTS)
     assert (len >= 0) $
 #endif
-    len <= 0
+    case len of
+      0 -> True
+      _ -> False
```

Comment:
`len == 0`? Does that change something compared to `len <= 0`?

Reply:
Yeah good point. I'll probably get rid of this change at some point. At the moment it makes Core diffs much easier to read

Reply:
Done in 53a52387e4faded50e1ded7d4a41eb6de5cd09ca